src/GraphQLParser/ParserContext.Parse.cs

```diff
@@ -248,21 +252,17 @@ private GraphQLDirectiveDefinition ParseDirectiveDefinition()
             ExpectKeyword("directive");
             Expect(TokenKind.AT);
 
-            var name = ParseName();
-            var args = ParseArgumentDefs();
-            bool repeatable = ParseRepeatable();
+            var def = NodeHelper.CreateGraphQLDirectiveDefinition(_ignoreOptions);
```

Comment:
I slightly changed all the `ParseXXX` methods:
```c#
  private GraphQLScalarValue ParseStringValue(/*bool isConstant*/)
        {
            var token = _currentToken;  // store token before node creation
                                                         // new line
            var val = NodeHelper.CreateGraphQLScalarValue(_ignoreOptions, ASTNodeKind.StringValue); // create node
                                                         // new line
            val.Comment = GetComment();    // store comment first
            Advance();                                     // some setup here
            val.Value = token.Value;
            val.Location = GetLocation(token.Start);
                                                                 // new line before return
            return val;
        }
```

I moved the initialization of node properties to a later stage (after node creation) so that it matches the grammar specification as much as possible (only in terms of reading).